pydicom/tests/test_handler_util.py

```diff
@@ -2016,3 +2019,32 @@ def test_short_bytestream(self):
         """Test result when no SIZ box present"""
         assert {} == get_j2k_parameters(b'')
         assert {} == get_j2k_parameters(b'\xff\x4f\xff\x51' + b'\x00' * 20)
+
+
+class TestGetNrFrames:
+    """Tests for get_nr_frames."""
+    def test_warning(self):
+        """Test warning when (0028,0008) 'Number of Frames' has a value of
+            None"""
+        ds = Dataset()
+        ds.NumberOfFrames = None
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
You can use:
```python
msg = (
    r"A value of None for (0028,0008) 'Number of Frames' is "
    r"non-conformant. It's recommended that this value be "
    r"changed to 1"
)
pytest.warns(UserWarning, match=msg):
    assert 1 == get_nr_frames(ds)
```

Reply:
`with pytest.warns` ...

Reply:
Whoops, fixed thanks

Reply:
Thanks, this is better. I'll update shortly.